---
title: >
  Using quantiseqr
author:
- name: Federico Marini^[marinif@uni-mainz.de]
  affiliation: Institute of Medical Biostatistics, Epidemiology and Informatics ([IMBEI, Mainz](https://www.unimedizin-mainz.de/imbei/imbei/welcome-page.html?L=1))
- name: Francesce Finotello^[ff@ff.ff]
  affiliation: ..., Innsbruck
date: "`r Sys.Date()`"
output: 
  rmarkdown::html_document:
    toc: true
    toc_float: true
    number_sections: true
    code_folding: show
    theme: lumen
bibliography: references_quantiseqr.bib
vignette: >
  %\VignetteIndexEntry{Using quantiseqr}
  %\VignettePackage{quantiseqr}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

<style type="text/css">
.smaller {
  font-size: 10px
}
</style>

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  # eval = FALSE,
  comment = "#>"
)
```

# Introduction {#introduction}

This vignette describes how to use the `r BiocStyle::Biocpkg("quantiseqr")` package for ... .

# Getting started {#gettingstarted}

To install this package, start R and enter:

```{r install, eval=FALSE}
if (!requireNamespace("BiocManager", quietly = TRUE))
  install.packages("BiocManager")

BiocManager::install("quantiseqr")
```

Once installed, the package can be loaded and attached to your current workspace as follows:

```{r loadlib, eval = TRUE}
library("quantiseqr")
```



```{r, message=FALSE, warning=FALSE}
library(dplyr)
library(ggplot2)
library(tidyr)
# library(immunedeconv)
library(tibble)
```

*Immunedeconv* ships with an example dataset with samples from four patients with metastatic melanoma published in [@EPIC2017]. 
It is available from `immunedeconv::dataset_racle`. It contains a gene expression matrix (`dataset_racle$expr_mat`) generated using bulk RNA-seq and 'gold standard' estimates of immune cell contents profiled with FACS (`dataset_racle$ref`). We are going to use the bulk RNA-seq data to run the deconvolution methods and will compare the results to the FACS data later on.   

The gene expression data is a matrix with HGNC symbols in rows and samples in columns:
The dataset is a matrix 

```{r}
# show the first 5 lines of the gene expression matrix
knitr::kable(dataset_racle$expr_mat[1:5, ])
```



To estimate immune cell fractions, we simply have to invoke the `deconvolute` function. It requires the specification
of one of the following methods for deconvolution:

```{r}
##### immunedeconv::deconvolution_methods
```

# Plattner et al analysis, example 1

Lorem ipsum description of the use case...

```{}
wget -c ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE107nnn/GSE107572/suppl/GSE107572%5Ftpm%5FPBMC%5FRNAseq%2Etxt%2Egz
unzip GSE107572_tpm_PBMC_RNAseq.txt.gz
read.table("GSE107572_tpm_PBMC_RNAseq.txt", header = TRUE)

```




```{r}
library(GEOquery)

tpmdata <-
  read.table("GSE107572_tpm_PBMC_RNAseq.txt", header = TRUE)

tpm_genesymbols <- tpmdata$GENE
tpmdata <- as.matrix(tpmdata[, -1])
rownames(tpmdata) <- tpm_genesymbols

ti_quant <- quantiseqr::run_quantiseq(
  expression_data = tpmdata,
  signature_matrix = "TIL10",
  is_arraydata = FALSE,
  is_tumordata = FALSE,
  scale_mRNA = TRUE
)

ti_quant

DCdata <- ti_quant


library(GEOquery)
GEOid <- "GSE107572"
gds <- getGEO(GEOid)
GEOinfo <- pData(gds[[1]])
FACSdata <- data.frame(
  B.cells = GEOinfo$`b cells:ch1`,
  T.cells.CD4 = GEOinfo$`cd4+ t cells:ch1`,
  T.cells.CD8 = GEOinfo$`cd8+ t cells:ch1`,
  Monocytes = GEOinfo$`monocytes:ch1`,
  Dendritic.cells = GEOinfo$`myeloid dendritic cells:ch1`,
  NK.cells = GEOinfo$`natural killer cells:ch1`,
  Neutrophils = GEOinfo$`neutrophils:ch1`,
  Tregs = GEOinfo$`tregs:ch1`
)
rownames(FACSdata) <- gsub("Blood-derived immune-cell mixture from donor ", "pbmc", GEOinfo$title) # Load deconvolution data obtained by running quanTIseq (--pipelinestart=decon)

# DCdata<-read.table(paste0(dir,â€/GSE107572_cell_fractions.txt"), header=TRUE, sep="\t", row.names=1)

rownames(DCdata) <- gsub("_.*$", "", sub("_", "", rownames(DCdata)))

ccells <- intersect(colnames(DCdata), colnames(FACSdata))
csbjs <- intersect(rownames(DCdata), rownames(FACSdata))

DCdata <- DCdata[csbjs, ccells]
FACSdata <- FACSdata[csbjs, ccells] # Compare cell fractions for single cell types and all cell types together
palette <- c("#451C87", "#b3b300", "#CE0648", "#2363C5", "#AB4CA1", "#0A839B", "#DD8C24", "#ED6D42")

names(palette) <- c("T.cells.CD4", "Dendritic.cells", "Monocytes", "T.cells.CD8", "Tregs", "B.cells", "NK.cells", "Neutrophils")
```


```{r fig.width=10, fig.height=10}
par(mfrow = c(3, 3))
colall <- c()
for (i in 1:(ncol(DCdata) + 1)) {
  if (i <= ncol(DCdata)) {
    x <- as.numeric(as.character(FACSdata[, i]))
    y <- DCdata[, i]
    ccell <- colnames(DCdata)[i]
    col <- palette[ccell]
  } else {
    x <- as.numeric(as.vector(as.matrix(FACSdata)))
    y <- as.vector(as.matrix(DCdata))
    ccell <- "All cells"
    col <- colall
  }
  res.cor <- cor.test(y, x)
  R <- round(res.cor$estimate, digits = 2)
  p <- format.pval(res.cor$p.value, digits = 2)
  RMSE <- round(sqrt(mean((y - x)^2, na.rm = TRUE)), digits = 2)

  regl <- lm(y ~ x)
  ymax <- max(round(max(y), digits = 2) * 1.3, 0.01)
  xmax <- max(round(max(x), digits = 2), 0.01)
  plot(x, y,
    main = gsub("(\\.)", " ", ccell), pch = 19,
    xlab = "Flow cytometry cell fractions",
    ylab = "quanTIseq cell fractions",
    col = col, cex.main = 1.3, ylim = c(0, ymax), xlim = c(0, xmax), las = 1
  )
  abline(regl)
  text(0, ymax * 0.98, cex = 1, paste0("r = ", R, ", p = ", p), pos = 4)
  text(0, ymax * 0.9, cex = 1, paste0("RMSE = ", RMSE), pos = 4)

  colall <- c(colall, rep(col, length(x)))
}
```


# Plattner et al, example 2

Lorem ipsum description of the use case...

```{r}
tpmdata <- read.table("tpm_GSE75299.txt", header = TRUE)

tpm_genesymbols <- tpmdata$GENE
tpmdata <- as.matrix(tpmdata[, -1])
rownames(tpmdata) <- tpm_genesymbols

ti_quant <- quantiseqr::run_quantiseq(
  expression_data = tpmdata,
  signature_matrix = "TIL10",
  is_arraydata = FALSE,
  is_tumordata = TRUE,
  scale_mRNA = TRUE
)

dim(ti_quant)
rownames(ti_quant)
colnames(ti_quant)
```


```{r fig.height=8, fig.width=8}
# download.file(
#   url = paste0("http://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?save=efetch&db=sra&rettype=runinfo&term=",
#                "SRP066571"),
#   destfile = "SRP066571runtable.txt",quiet = TRUE)

library(GEOquery)
library(ggplot2)
library(reshape2)

### Load file with estimated cell fractions by quanTIseq:
cdata <- # read.csv(paste0(path, "/SRP066571_cell_fractions.txt"), header = TRUE, sep = "\t", stringsAsFactors = FALSE, row.names = 1)
  ti_quant


### Annotation
gds <- getGEO("GSE75299")
pdata <- pData(gds[[1]])

sraInfo <- read.csv("SRP066571runtable.txt", header = TRUE, sep = ",", stringsAsFactors = FALSE)

cdata <- cbind(cdata, sraInfo[, "SampleName"][match(rownames
(cdata), sraInfo$Run)])

colnames(cdata)[length(cdata)] <- "geo_accession"
dim(pdata)
pdata <- pdata[grep("^Pt", pdata[, 1]), ]

pt_matchedinfo <- pdata[, "title"][match(cdata$geo_accession, pdata$geo_accession)]
cdata$pt_info <- pt_matchedinfo
# rownames(cdata) <- cdata[, length(cdata)]

cdata$groups <- "POST"
cdata$groups[grep("baseline", cdata$pt_info)] <- "PRE"

class(cdata)

cdata

library(tidyr)
cellfracs <- pivot_longer(cdata, cols = B.cells:Other)

cellfracs$name <- factor(cellfracs$name,
                         levels = c("B.cells", "Macrophages.M1", "Macrophages.M2",
                                    "Monocytes", "Neutrophils", "NK.cells",   
                                    "T.cells.CD4", "T.cells.CD8", "Tregs", 
                                    "Dendritic.cells", "Other"))

ggplot(cellfracs, aes(fill = name, y = value, x = Sample)) + 
  geom_bar(position = "fill", stat = "identity") + 
  scale_fill_brewer(palette = "PuOr") + 
  # viridis::scale_fill_viridis(discrete = TRUE) +
  xlab("") + 
  ylab("Cell Fractions") + 
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 45, hjust=1))


prepost_data <- cdata[!is.na(cdata$pt_info),]
prepost_data <- prepost_data[-grep("Other", colnames(prepost_data))]

### Remove other cells and the last two columns

# cdata <- cdata[, -grep("Other", colnames(cdata))]

# cdata <- cdata[, -grep("geo_accession", colnames(cdata))]

prepost_data <- pivot_longer(prepost_data, cols = B.cells:Dendritic.cells)

prepost_data$groups <- factor(prepost_data$groups, levels = c("PRE", "POST"))

prepost_data$name <- factor(prepost_data$name,
                            levels = c("B.cells", "Macrophages.M1", "Macrophages.M2",
                                       "Monocytes", "Neutrophils", "NK.cells",   
                                       "T.cells.CD4", "T.cells.CD8", "Tregs", 
                                       "Dendritic.cells"))

ggplot(prepost_data, aes(name, value, fill = groups)) +
  geom_boxplot() +
  xlab("") + 
  ylab("cell fractions") +
  theme_bw() + 
  theme(axis.text.x = element_text(angle = 45, hjust=1)) 
# +
#   stat_compare_means(aes(group = groups),
#     symnum.args = list(cutpoints = c(0, 0.001, 0.01, 0.05, 0.1, 1), symbols = c("****", "***", "**", "*", "ns")),
#     label = "p.signif",
#     hide.ns = TRUE, cex = 10
#   ) # default: Wilcoxon Test
```


# General calls and tests 

to see that everything returns the same proportions...



For this example, we use `quanTIseq`. As a result, we obtain a cell_type x sample data frame with cell-type scores for each sample.

this is not the way we should call it, but let it be handled via `immunedeconv`

```{r, message=FALSE, warning=FALSE}
# res_quantiseq <- quantiseqr::deconvolute_quantiseq(
#   dataset_racle$expr_mat,
#   tumor = TRUE,
#   arrays = FALSE,
#   scale_mrna = TRUE
# )
```

in there for checking vs `run_quantiseq`

```{r}
run_tests <- FALSE
```


```{r, message=FALSE, warning=FALSE, eval=run_tests}
res_quantiseq_immunedeconv <- immunedeconv::deconvolute(dataset_racle$expr_mat, "quantiseq", tumor = TRUE)
```

This one would be the way we should use to call it!

```{r, message=FALSE, warning=FALSE}
res_quantiseq_run <- quantiseqr::run_quantiseq(
  expression_data = dataset_racle$expr_mat,
  signature_matrix = "TIL10",
  is_arraydata = FALSE,
  is_tumordata = TRUE,
  scale_mRNA = TRUE
)

sample_names <- res_quantiseq_run$Sample
res_mat <- res_quantiseq_run %>%
  as_tibble() %>%
  select(-Sample) %>%
  as.matrix()
rownames(res_mat) <- sample_names

res_mat <- t(res_mat)
colnames(res_mat) <- rownames(res_quantiseq_run)
```

<!-- some checks here... -->

<!-- ```{r, message=FALSE, warning=FALSE, eval=run_tests} -->
<!-- # identical(res_mat, res_quantiseq) -->

<!-- all.equal(as.matrix(res_quantiseq_immunedeconv[, -1]), res_mat) -->
<!-- as.matrix(res_quantiseq_immunedeconv[, -1]) - res_mat -->
<!-- ``` -->


QuanTIseq generates scores that can be interpreted as a cell-type fraction. Let's visualize
the results as a stacked bar chart with tidyverse/ggplot2.

```{r, fig.height=4, fig.width=8}
res_mat %>%
  as.data.frame() %>%
  rownames_to_column(var = "cell_type") %>%
  gather(sample, fraction, -cell_type) %>%
  # plot as stacked bar chart
  ggplot(aes(x = sample, y = fraction, fill = cell_type)) +
  geom_bar(stat = "identity") +
  coord_flip() +
  scale_fill_brewer(palette = "Paired") +
  scale_x_discrete(limits = rev(levels(res_mat)))
```

We observe that

* the first two samples (LAU355, LAU1314) appear to contain a large amount of CD4+ T cells and B cells
* the last two samples (LAU1255, LAU125) appear to contain a large amount of "uncharacterized cells"
* the last sample (LAU125) appears to contain no CD8+ T cells, often associated with bad prognosis. 

Estimating the amount of "uncharacterized cells" is a novel feature introduced by quanTIseq and EPIC [@EPIC2017, @quantiseq2019]. This estimate often corresponds to the fraction of tumor cells in the sample. 





<!-- Let's now apply MCP-counter to the same dataset. -->

<!-- ```{r, message=FALSE, warning=FALSE} -->
<!-- library(immunedeconv) -->
<!-- res_mcp_counter <- deconvolute(dataset_racle$expr_mat, "mcp_counter") -->
<!-- ``` -->

<!-- MCP-counter provides scores in arbitrary units that are only comparable between samples, but not between -->
<!-- cell-types. The visualization as bar-chart suggests the scores to be cell-type fractions and is, therefore, unsuitable. -->
<!-- Instead, we use ggplot to visualize the scores per-cell type, allowing for a relative comparison between samples. -->

<!-- ```{r, fig.height=7, fig.width=7} -->
<!-- res_mcp_counter %>% -->
<!--   gather(sample, score, -cell_type) %>% -->
<!--   ggplot(aes(x = sample, y = score, color = cell_type)) + -->
<!--   geom_point(size = 4) + -->
<!--   facet_wrap(~cell_type, scales = "free_x", ncol = 3) + -->
<!--   scale_color_brewer(palette = "Paired", guide = FALSE) + -->
<!--   coord_flip() + -->
<!--   theme_bw() + -->
<!--   theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) -->
<!-- ``` -->

<!-- With the scores being in arbitrary units, the results are not useful for judging if a cell type is present in the sample, or not. However, we can compare the relative values between samples and relate them to the results we obtained earlier using quanTIseq.  -->

<!-- * Consistent with quanTIseq, MCP-counter predicts B cells to be more abundant in LAU355 and LAU1314 than in the other two samples.  -->
<!-- * Also consistent with quanTIseq, the last LAU125 appears to contain significantly less CD8+ T cells than the other samples.  -->
<!-- * the estimates for Neutrophils and Monocytes are inconsistent with quanTIseq.  -->


## Comparison with FACS data

Let's now compare the results with 'gold standard' FACS data obtained for the four samples. This is, of course, not a representative benchmark, but it gives a notion about what magnitude of predictive accuracy we can expect. 

...

# Additional Information {#additionalinfo}

e.g. using quantiseqr via immunedeconv
...

# FAQs {#faqs}

**Q: MyQ?**

A: MyA

# Session Info {- .smaller}

```{r sessioninfo}
sessionInfo()
```



# References {-}
